### ISSUE TITLE
Add support for teaching clj-kondo how to lint custom tagged literals

### ISSUE THREAD

**Issue by didibus** [clj-kondo/clj-kondo#1127]
**Is your feature request related to a problem? Please describe.**

Imagine someone creates a tagged literal like so:

```
#fn(a b | + a b) ;=> (fn [a b] (+ a b))
```

Now say they install it into their `data_readers.clj`, or they bind it to `*data-readers*`, meaning they will start using it in their code. The issue is, clj-kondo will not know how to lint this tag literal form, and will start reporting all kinds of wrong linter warnings and errors.

**Describe the solution you'd like**

A way to teach clj-kondo how to lint custom tagged literals, possibly similar to how you can teach it to lint macros using lint-as and hooks.

**Describe alternatives you've considered**

While I'd rather clj-kondo can be thought how to lint custom tagged literals. A quick solution would be a way to tell clj-kondo to just ignore linting certain custom tagged-literals, so at least you don't get false lint errors and warnings.

**Comment by borkdude** [comment]
@didibus Right now clj-kondo "ignores" unresolved symbols inside reader tags. Are you still interested in this feature, e.g. something like `hooks {:analyze-tagged-literal {fn ...}}` (analogs to `:analyze-call` or `:macroexpand-tagged-literal` analogous to `:macroexpand`?

**Comment by borkdude** [comment]
I'm interested in this myself since clavascript and cherry have a #jsx data reader and currently unresolved symbols in expressions are ignored. I think it should just lint as the identity function.

**Comment by didibus** [comment]
I don't use tagged literal very often, so ignoring them is a good quick solution. It would still be nice to be able to teach it how to lint them, similar a bit to how it can be thought to lint macros. But I wouldn't put it high on my priority list.
